cli/src/error.rs

```diff
@@ -1,30 +1,40 @@
 use super::test_highlight;
 use std::fmt::Write;
 use std::io;
+use std::io::ErrorKind;
 use tree_sitter::{QueryError, QueryErrorKind};
 use walkdir;
 
 #[derive(Debug)]
-pub struct Error(pub Vec<String>);
+pub struct Error(Option<Vec<String>>);
 
 pub type Result<T> = std::result::Result<T, Error>;
 
 impl Error {
     pub fn grammar(message: &str) -> Self {
-        Error(vec![format!("Grammar error: {}", message)])
+        Error(Some(vec![format!("Grammar error: {}", message)]))
     }
 
     pub fn regex(mut message: String) -> Self {
         message.insert_str(0, "Regex error: ");
-        Error(vec![message])
+        Error(Some(vec![message]))
     }
 
     pub fn undefined_symbol(name: &str) -> Self {
-        Error(vec![format!("Undefined symbol `{}`", name)])
+        Error(Some(vec![format!("Undefined symbol `{}`", name)]))
     }
 
     pub fn new(message: String) -> Self {
-        Error(vec![message])
+        let e = Error(Some(vec![message]));
+        e
```

Comment:
Looks like you could eliminate this `let`.

Reply:
Done, I've forgot it after debugging while assigning to a var forces rust-analyzer to show derived type.